docs/DEVELOPMENT.md

```diff
@@ -112,57 +112,101 @@ Some of the e2e jobs are not triggered automatically. We recommend to trigger th
 
 >Note: only members with commit permission in the repository are allowed to trigger the e2e jobs. If you are not a committer then ask for help on our main Slack channel (#confidential-containers).
 
-### Running e2e tests on your local machine
+## Running CI e2e tests on your local machine
 
 We recommend that you run the e2e Non-TEE tests on your local machine before opening a PR to check your changes will not break the CI so to avoid wasting resources. You can also use the approach described below to debug and fix failures, or test changes on the scripts themselves.
 
-The entry point script is [tests/e2e/run-local.sh](../tests/e2e/run-local.sh). It is going to install softwares and change the system's configuration, so we recommend that you run the e2e tests on VMs with nested virtualization support and a minimum of 8GB of memory, 50 GB of disk and 4 vCPUs.
+There are three main ways of running the tests locally:
 
-Currently the e2e tests are supported on Ubuntu 20.04 or CentOS 8 Stream, and the only requirement is to have Ansible installed.
+1. use vagrant script to run the full suite
+2. use kcli to create a VM where you run the testing
+3. running them directly on your development workstation
 
-For example, to run on a fresh Ubuntu 20.04 VM:
+### Using vagrant:
+
+This is the simplest method but for each invocation it builds everything, which might take about 50m for the first time (then about 30m each). To perform this check install [Vagrant](https://www.vagrantup.com) and run the [Vagrantfile](../tests/e2e/Vagrantfile) to perform it's task (optionally setting the RUNTIMECLASS):
 
 ```shell
+cd tests/e2e
+export RUNTIMECLASS="kata-qemu"
+vagrant up tests-e2e-ubuntu2204
+```
+
+### Using kcli
+
+You can leverage [kcli](https://github.com/karmab/kcli/) to provide and maintain your VM that can be used for testing. Some useful commands:
+
+```shell
+# Create a machine compatible with our testing
+kcli create vm -i ubuntu2204 -P memory=8G -P numcpus=4 -P disks=[50] e2e
+# Ssh to the machine
+kcli ssh e2e
+# Stop the machine
+kcli stop vm e2e
+# Start the machine
+kcli start vm e2e
```

Comment:
nit: unless stopping and starting are mandatory here, I would suggest removing stop and start here. We could redirect users to full kcli documentation if necessary to learn more.

Reply:
My intention was to provide a subset of commands that are likely to be used by users when testing coco op (it's written in the heading "Some useful commands"). So I'd rather keep it here but if you insist I can remove it.

Reply:
I believe those type of instructions will make the document a little bigger and forces the contributor to be selective about what should be coping and pasting during the tests.

If we left here only the essential commands, it would be better. The tutorial could be on how to reproduce the test and not on how to use kcli.  We could point readers to the official page, reducing our chances of getting those commands deprecated (in case of a future syntax change).

Reply:
Fair enough, let me simplify it a bit :-)